cmake/pcre2-config.cmake.in

```diff
@@ -23,70 +27,33 @@
 #   PCRE2::32BIT - The 32 bit PCRE2 library.
 #   PCRE2::POSIX - The POSIX PCRE2 library.
 
-set(PCRE2_NON_STANDARD_LIB_PREFIX @NON_STANDARD_LIB_PREFIX@)
-set(PCRE2_NON_STANDARD_LIB_SUFFIX @NON_STANDARD_LIB_SUFFIX@)
-set(PCRE2_8BIT_NAME pcre2-8)
-set(PCRE2_16BIT_NAME pcre2-16)
-set(PCRE2_32BIT_NAME pcre2-32)
-set(PCRE2_POSIX_NAME pcre2-posix)
-find_path(PCRE2_INCLUDE_DIR NAMES pcre2.h DOC "PCRE2 include directory")
-if (PCRE2_USE_STATIC_LIBS)
-  if (MSVC)
-    set(PCRE2_8BIT_NAME pcre2-8-static)
-    set(PCRE2_16BIT_NAME pcre2-16-static)
-    set(PCRE2_32BIT_NAME pcre2-32-static)
-    set(PCRE2_POSIX_NAME pcre2-posix-static)
-  endif ()
+@PACKAGE_INIT@
 
-  set(PCRE2_PREFIX ${CMAKE_STATIC_LIBRARY_PREFIX})
-  set(PCRE2_SUFFIX ${CMAKE_STATIC_LIBRARY_SUFFIX})
-else ()
-  set(PCRE2_PREFIX ${CMAKE_SHARED_LIBRARY_PREFIX})
-  if (MINGW AND PCRE2_NON_STANDARD_LIB_PREFIX)
-    set(PCRE2_PREFIX "")
-  endif ()
-
-  set(PCRE2_SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
-  if (MINGW AND PCRE2_NON_STANDARD_LIB_SUFFIX)
```

Comment:
how is this PCRE2_NON_STANDARD_LIB_SUFFIX option handled in the new setup?

Reply:
The names of the libraries are hardcoded in the generated export target files. Here's a snippet of the generated file:

```cmake
# Import target "pcre2::pcre2-8-shared" for configuration "Debug"
set_property(TARGET pcre2::pcre2-8-shared APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)
set_target_properties(pcre2::pcre2-8-shared PROPERTIES
  IMPORTED_IMPLIB_DEBUG "${_IMPORT_PREFIX}/lib/pcre2-8d.lib"
  IMPORTED_LOCATION_DEBUG "${_IMPORT_PREFIX}/bin/pcre2-8d.dll"
  )

list(APPEND _cmake_import_check_targets pcre2::pcre2-8-shared )
list(APPEND _cmake_import_check_files_for_pcre2::pcre2-8-shared "${_IMPORT_PREFIX}/lib/pcre2-8d.lib" "${_IMPORT_PREFIX}/bin/pcre2-8d.dll" )
```

And here's with a non-standard suffix specified:

```cmake
# Import target "pcre2::pcre2-8-shared" for configuration "Debug"
set_property(TARGET pcre2::pcre2-8-shared APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)
set_target_properties(pcre2::pcre2-8-shared PROPERTIES
  IMPORTED_IMPLIB_DEBUG "${_IMPORT_PREFIX}/lib/pcre2-8d.lib"
  IMPORTED_LOCATION_DEBUG "${_IMPORT_PREFIX}/bin/pcre2-8d-0.dll"
  )

list(APPEND _cmake_import_check_targets pcre2::pcre2-8-shared )
list(APPEND _cmake_import_check_files_for_pcre2::pcre2-8-shared "${_IMPORT_PREFIX}/lib/pcre2-8d.lib" "${_IMPORT_PREFIX}/bin/pcre2-8d-0.dll" )
```

Reply:
Done.